pandas/_libs/interval.pyx

```diff
@@ -107,6 +107,59 @@ cdef class IntervalMixin:
         """Return the length of the Interval"""
         return self.right - self.left
 
+    @property
+    def is_empty(self):
+        """
+        Indicates if an interval is empty, meaning it contains no points.
+
+        .. versionadded:: 0.25.0
+
+        Returns
+        -------
+        bool or ndarray
+            A boolean indicating if a scalar :class:`Interval` is empty, or a
+            boolean ``ndarray`` positionally indicating if an ``Interval`` in
+            an :class:`~arrays.IntervalArray` or :class:`IntervalIndex` is
+            empty.
+
+        Examples
+        --------
+        An :class:`Interval` that contains points is not empty:
+
+        >>> pd.Interval(0, 1, closed='right').is_empty
+        False
+
+        An ``Interval`` that does not contain any points is empty:
+
+        >>> pd.Interval(0, 0, closed='right').is_empty
+        True
+        >>> pd.Interval(0, 0, closed='left').is_empty
+        True
+        >>> pd.Interval(0, 0, closed='neither').is_empty
+        True
+
+        An ``Interval`` that contains a single point is not empty:
+
+        >>> pd.Interval(0, 0, closed='both').is_empty
+        False
+
+        An :class:`~arrays.IntervalArray` or :class:`IntervalIndex` returns a
+        boolean ``ndarray`` positionally indicating if an ``Interval`` is
+        empty:
+
+        >>> ivs = [pd.Interval(0, 0, closed='neither'),
+        ...        pd.Interval(1, 2, closed='neither')]
+        >>> pd.core.arrays.IntervalArray(ivs).is_empty
```

Comment:
nitpick: this should be `pd.IntervalArray(ivs).is_empty`

Reply:
`IntervalArray` isn't top level:
```python
In [1]: import pandas as pd; pd.__version__
Out[1]: '0.25.0.dev0+894.g31d93e5d9'

In [2]: pd.IntervalArray
---------------------------------------------------------------------------
AttributeError: module 'pandas' has no attribute 'IntervalArray'
```